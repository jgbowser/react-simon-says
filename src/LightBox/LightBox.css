.light_box {
  display: inline-block;
  width: 125px;
  height: 125px;
  margin: 10px;
  border-radius: 10px;
}