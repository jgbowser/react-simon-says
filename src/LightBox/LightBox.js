import React from 'react'
import './LightBox.css'

export default class LightBox extends React.Component {
  static defaultProps = {
    color: 'white'
  }
  render() {
    return (
      <div className="light_box" style={{backgroundColor: this.props.color}}></div>
    )
  }
}