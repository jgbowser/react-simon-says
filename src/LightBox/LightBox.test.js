import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import LightBox from './LightBox'

describe('LightBox Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<LightBox />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('should render as white without any props passed in', () => {
    const tree = renderer
      .create(<LightBox />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render as green when green is passed in as a prop', () => {
    const tree = renderer
      .create(<LightBox color="green" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
