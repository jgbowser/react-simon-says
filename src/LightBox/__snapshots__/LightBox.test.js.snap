// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightBox Component should render as green when green is passed in as a prop 1`] = `
<div
  className="light_box"
  style={
    Object {
      "backgroundColor": "green",
    }
  }
/>
`;

exports[`LightBox Component should render as white without any props passed in 1`] = `
<div
  className="light_box"
  style={
    Object {
      "backgroundColor": "white",
    }
  }
/>
`;
